### PR TITLE
haddock-combine: Include dependencies of hsdocs so all haddock links work

### PR DESCRIPTION
**Pull request by shlevy**
**Please do a proper merge, _not_ a squash/rebase, when merging. The parent commit of this PR is intentional**

### CODE REVIEW

**Comment by shlevy** [comment]
Found a bug, temporarily converting to draft.

**Comment by shlevy** [comment]
Working now.

File: nix/lib/haddock-combine.nix [hunk 1]
```diff
@@ -7,14 +7,24 @@
 let
   hsdocs = builtins.map (x: x.doc) (builtins.filter (x: x ? doc) hspkgs);
 in
-runCommand "haddock-join" { buildInputs = [ hsdocs ]; } ''
-  # Merge all the docs from the packages. We don't use symlinkJoin because:
+runCommand "haddock-join"
+{
+  buildInputs = [ hsdocs ];
+  exportReferencesGraph = lib.concatLists
```
Comment (michaelpj) [inline nix/lib/haddock-combine.nix:13]: Seems like it needs a comment. What's going on here? Something clever for sure.

**Comment by michaelpj** [comment]
> The parent commit of this PR is intentional

Why? What are you trying to do?

File: nix/lib/haddock-combine.nix (hunk 1 above)
Comment (shlevy) [reply nix/lib/haddock-combine.nix:18]: Added

**Comment by shlevy** [comment]
> > The parent commit of this PR is intentional
> 
> Why? What are you trying to do?

I want to be able to use these changes in `marlowe-cardano`, which is on an older version of `plutus`. This will allow us to directly use cb126ff6d3f20985b34cf19ca31fc7f80f3ae550 instead of creating a backport branch with a cherry-pick

**Comment by shlevy** [comment]
> Comment line lengths are too long, see hydra.

I get line too long even if I put the Nix manual URL on its own line, is there any way to disable it for that line?

**Comment by michaelpj** [comment]
Like so: https://github.com/input-output-hk/plutus/blob/master/nix/pkgs/haskell/extra.nix#L73

Or you  could disable it for the file, not a big deal.

**Pull request merged**